### PR TITLE
ENG-8435: Automated docs

### PR DESCRIPTION
**Pull request by wcmjunior**
## Description of the change

Standardize documentation creation through [`terraform-plugin-docs`](https://github.com/hashicorp/terraform-plugin-docs) and adds features to allow the documentation to be built from the source code.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

All unit tests passed successfully.

### CODE REVIEW

File: cyral/resource_cyral_repository_identity_map.go [hunk 1]
```diff
@@ -199,6 +199,7 @@ var ReadRepositoryIdentityMapConfig = ResourceOperationConfig{
 
 func resourceRepositoryIdentityMap(deprecationMessage string) *schema.Resource {
 	return &schema.Resource{
+		Description:        "Manages [Repository Identity Maps] configuration(https://cyral.com/docs/manage-repositories/repo-id-map/).",
```
Comment (VictorGFM) [inline cyral/resource_cyral_repository_identity_map.go:202]:
```suggestion
		Description:        "Manages [Repository Identity Maps](https://cyral.com/docs/manage-repositories/repo-id-map/) configuration.",
```

**Review by VictorGFM — COMMENTED** [review]
Looks really great, automating the docs will definitely make our lives easier and will help us provide more robust docs. I have a comment about the pre-commit tool and a few minor comments regarding typos and details though.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,24 @@ terraform {
...
+for more information on where and how to store examples and define templates for documentation
+artifacts.
+
+To create the documentation automatically, run the commands:
+
+```bash
+# Creates the documentation files from the source code
+make docker-compose/docs
+# Runs the pre-commit linter
+pre-commit run --show-diff-on-failure --color=always --all-files
```
[the remark this answers is not in the document]
Comment (wcmjunior) [reply README.md:60]: The repository is back online. @greenkiwi will keep us posted if they make any change to the organization.

File: Makefile [hunk 3]
```diff
@@ -70,3 +76,7 @@ docker-compose/build: docker-compose/lint
 
 docker-compose/lint:
 	docker-compose run lint
+
+docker-compose/docs:
+	docker-compose run app go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs generate
+#docker-compose run build pre-commit run --show-diff-on-failure --color=always --all-files
```
[the remark this answers is not in the document]
Comment (wcmjunior) [reply Makefile:82]: I tried to do that, but wasn't able to. Let's run it locally for now and create a separate ticket for that.

**Review by VictorGFM — APPROVED** [review]
Great work :)

**Pull request merged**